util.py

```diff
@@ -0,0 +1,12 @@
+def convert_str_to_bool(value: str, default: bool) -> bool:
+    try:
+        if value is None:
+            return default
+        if value.lower() == 'true':
+            return True
+        elif value.lower() == 'false':
+            return False
+        else:
+            return default
+    except Exception as e:
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider specific exception handling</b></div>
<div id="fix">

The `except` block catches all exceptions without any specific handling. Consider catching specific exceptions like `AttributeError` that could occur when calling `lower()` on non-string values.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -1,12 +1,14 @@
 def convert_str_to_bool(value: str, default: bool) -> bool:
     try:
         if value is None:
             return default
         if value.lower() == 'true':
             return True
         elif value.lower() == 'false':
             return False
         else:
             return default
 -    except Exception as e:
 +    except AttributeError:
 +        return default
 +    except ValueError:
         return default
```

```suggestion
    except AttributeError:
        return default
    except ValueError:
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/rkj6124/test/pull/2#issuecomment-2482623252>#103e48</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged